### PR TITLE
Add GHA workflow to run lint and tests

### PR DESCRIPTION
**Pull request by mfiedorowicz**
Adding GitHub Actions workflow to run lint (with `ruff`), building docs (`mkdocs build`) and tests

### CODE REVIEW

File: .github/lint-tests.yml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  checks: write
+  pull-requests: write
+
+jobs:
+  tests:
+    runs-on: ubuntu-latest
+    timeout-minutes: 10
+    strategy:
+      matrix:
+        python: [ "3.10" ]
```
Comment (mfiedorowicz) [inline .github/lint-tests.yml]: Starting with `3.10` for time being, we can expand it with `3.11` and `3.12` once workflow is working properly.

File: pyproject.toml [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+]
+package-data = {"netbox_branching" = ["**/*", "templates/**"]}
+exclude-package-data = {netbox_branching = ["tests/*"]}
+license-files = ["LICENSE.md"]
+
+[build-system]
+requires = ["setuptools>=43.0.0", "wheel"]
+build-backend = "setuptools.build_meta"
+
+[tool.ruff]
```
Comment (jeremystretch) [inline pyproject.toml]: Let's disable the checks for docstrings to cut down on noise. (These sometimes aren't needed, and their formatting is standard.)

File: .github/workflows/lint-tests.yml [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+  pull-requests: write
+
+jobs:
+  tests:
+    runs-on: ubuntu-latest
+    timeout-minutes: 10
+    strategy:
+      matrix:
+        python: [ "3.10" ]
+    steps:
```
Comment (jeremystretch) [inline .github/workflows/lint-tests.yml]: We should include a step that builds the documentation using `mkdocs build` as well.

File: pyproject.toml (hunk 2 above)
Comment (mfiedorowicz) [reply pyproject.toml]: Disabled ~most~ all checks for `missing docstrings`.

File: .github/workflows/lint-tests.yml (hunk 3 above)
Comment (mfiedorowicz) [reply .github/workflows/lint-tests.yml]: Where should we build into?

Comment (mfiedorowicz) [reply .github/workflows/lint-tests.yml]: Leaving in the default location for now (`./site`)

**Comment by jeremystretch** [comment]
I'm concerned about the introduction of a Docker container for testing and all the additional complexity it brings. Does this buy us anything that we wouldn't get with a [much simpler workflow](https://github.com/peteeckel/netbox-plugin-dns/blob/main/.github/workflows/test.yml) running NetBox natively?

**Comment by mfiedorowicz** [comment]
> I'm concerned about the introduction of a Docker container for testing and all the additional complexity it brings. Does this buy us anything that we wouldn't get with a [much simpler workflow](https://github.com/peteeckel/netbox-plugin-dns/blob/main/.github/workflows/test.yml) running NetBox natively?

Agreed, especially that we can run tests against various python versions.

**Review by jeremystretch — APPROVED** [review]

**Pull request merged**